src/lang.rs

```diff
@@ -884,6 +909,81 @@ impl Context for CallStack {
         }
     }
 
+    fn get_mutable(&mut self, name: String) -> EvalResult {
+        let mut env = self.env();
+        let mut caller_env = true;
+        loop {
```

Comment:
Probably not going to be able to fully finish review today, so this is just a note to myself to revisit this loop.